src/swh_api_functions.py

```diff
@@ -0,0 +1,50 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# heritedcode is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/nexB/heritedcode for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+
+import requests
+
+class CustomApiResponse:
+    """
+    this class represents response data structure for returning response
+    from functions.
+    """
+
+    def __init__(self, data):
+        self.checksums = None
```

Comment:
@pombredanne I have a question here, as we will be having lot of attributes from swh and some of them will be present in some api's response and some won't be. But as you said we need to pass each and every attribute as an arg, so if we dont have any attribute in some api response thats not an error right?? we'll need to pass None int that case.
So basically from my point of view, passing each attribute as an arg seems little tedious instead of a dict/json. Every time i pass arg i'll have to check its sequence in constructor, and even if any arg is missing from total args in class that does not mean its an error because every api call will have diff set of attributes in response not all of'em will be present every time.

Whats your take on this??Hope you got me??
Thanks!! :)

Reply:
> But as you said we need to pass each and every attribute as an arg, so if we dont have any attribute in some api response thats not an error right??

That's an error alright. Or that's just return None, but that's not something that can be ignored silently, 

> So basically from my point of view, passing each attribute as an arg seems little tedious instead of a dict/json. 

Tedious but that's what is correct... we have  concrete data with a schema and the schema exists always  :) ... that's also why there are utilities like https://www.attrs.org that remove a lot of the gruntwork of the boilerplate.
In a way to me, applying some schemata to some external data is part of the essence of programming: you apply order to chaos and apply intelligence and structure to understand the external world. 

To may be better understand why you need concrete attributes vs. a plain schemaless dict, please view this https://martinfowler.com/tags/noSQL.html and https://martinfowler.com/articles/schemaless/ and this video http://www.youtube.com/watch?v=qI_g07C_Q5I


>  Every time i pass arg i'll have to check its sequence in constructor, and even if any arg is missing from total args in class that does not mean its an error because every api call will have diff set of attributes in response not all of'em will be present every time.

That's why you need a factory that will pick the field you support.

Reply:
Okay..Got it..
Thanks.. :)